ftw/noticeboard/content/notice.py

```diff
@@ -14,9 +17,32 @@ class INotice(Interface):
     """Marker interface for the Notice"""
 
 
+def user_mail():
+    if not api.user.is_anonymous():
+        return safe_unicode(api.user.get_current().getProperty('email'))
+    return u''
+
+
 class INoticeSchema(model.Schema):
     """A Folderish type for notices, which may hold images"""
 
+    price = schema.TextLine(
+        title=_(u'label_price', default=u'Price'),
+        required=True,
+    )
+
+    email = schema.TextLine(
+        title=_(u'label_email', default=u'E-Mail'),
+        required=True,
+        defaultFactory=user_mail,
```

Comment:
Why doesn't that have to be called?

Reply:
It expects a `factory`, respectively a `callable`.  Since the schema is loaded on boot time, there is no user available at this point. `defaultFactory` defers setting the actual default value to the form initialisation. You can use the param `default` If you want a kinda static default value. 

Further you can also register a context aware default factory, which makes a context and request available for figuring out the default value. 

For more infos, see https://docs.plone.org/external/plone.app.dexterity/docs/advanced/defaults.html

Reply:
Awesome thanks!